src/main/java/org/veupathdb/service/eda/ss/model/reducer/StreamUnionMerger.java

```diff
@@ -74,6 +83,17 @@ private void setNext() {
     next = nextElement.idIndex;
   }
 
+  @Override
+  public void close() throws Exception {
```

Comment:
Here and above; a while back I wrote a closeable list clase for this use case.  Not sure if it's completely compatible with how you're using these.  It's nice because you can try-with-resources on the List.  Maybe take a look and see; could save you a loop or two.  Looks like we bury exceptions though.
https://github.com/VEuPathDB/FgpUtil/blob/master/Core/src/main/java/org/gusdb/fgputil/AutoCloseableList.java